packages/blade/src/components/Input/PhoneNumberInput/PhoneNumberInput.web.tsx

```diff
@@ -0,0 +1,216 @@
+
+  const handleOnChange = ({
+    name,
+    value,
+    selectedCountry,
+  }: {
+    name?: string;
+    value?: string;
+    selectedCountry: CountryCodeType;
+  }): void => {
+    onChange?.({
+      name: name!,
+      value: value!,
+      phoneNumber: value ? formatPhoneNumber(value, selectedCountry) : undefined!,
+      dialCode: getDialCodeByCountryCode(selectedCountry),
+      countryCode: selectedCountry,
+    });
+  };
+
+  const onItemClick = ({ name }: { name: string }): void => {
+    const selectedCountry = name as CountryCodeType;
+    setSelectedCountry(selectedCountry);
+    handleOnChange({
+      selectedCountry,
+      name: inputRef.current?.name,
+      value: inputRef.current?.value,
+    });
+    inputRef.current?.focus();
+  };
+
+  return (
+    <BaseInput
+      setInputWrapperRef={(node) => {
+        inputWrapperRef.current = node as HTMLInputElement;
+      }}
+      ref={mergedRef}
+      id="textinput"
+      componentName="PhoneNumberInput"
+      label={label as string}
+      hideLabelText={!Boolean(label)}
+      labelPosition={labelPosition}
+      defaultValue={defaultValue}
+      value={value}
+      name={name}
+      placeholder={formatPhoneNumber('1234567890', selectedCountry)}
+      trailingIcon={trailingIcon}
+      leadingIcon={leadingIcon}
+      prefix={showDialCode ? getDialCodeByCountryCode(selectedCountry) : undefined}
+      trailingInteractionElement={renderTrailingInteractionElement()}
+      leadingInteractionElement={
+        showCountrySelector ? (
+          <CountrySelector
+            size={size}
+            countryData={countryData}
+            flags={flags}
+            inputWrapperRef={inputWrapperRef}
+            isDisabled={isDisabled}
+            selectedCountry={selectedCountry}
+            onItemClick={onItemClick}
+          />
+        ) : null
+      }
+      onChange={({ name, value }) => {
+        if (value?.length) {
+          // show the clear button when the user starts typing in
+          setShouldShowClearButton(true);
+        }
+        if (shouldShowClearButton && !value?.length) {
+          // hide the clear button when the input field is empty
+          setShouldShowClearButton(false);
+        }
+        handleOnChange({ name, value, selectedCountry });
+      }}
+      onClick={onClick}
+      onFocus={onFocus}
+      onBlur={onBlur}
+      isDisabled={isDisabled}
+      accessibilityLabel={accessibilityLabel}
+      necessityIndicator={necessityIndicator}
+      isRequired={isRequired}
+      validationState={validationState}
+      errorText={errorText}
+      helpText={helpText}
+      successText={successText}
+      // eslint-disable-next-line jsx-a11y/no-autofocus
+      autoFocus={autoFocus}
+      testID={testID}
+      size={size}
+      {...getKeyboardAndAutocompleteProps({
+        type: 'number',
+        keyboardReturnKeyType,
+        autoCompleteSuggestionType,
+        autoCapitalize,
+      })}
+      type="number"
```

Comment:
shouldn't it be `type="telephone"` 🤔

Reply:
Yup I think. Because of type="number" HTML's default number things are coming up like pressing up and down arrows increases and decreases the counter
<img width="318" alt="image" src="https://github.com/razorpay/blade/assets/30949385/31f8a971-98a7-4f35-81c3-083a30d87cd1">

Reply:
`telephone` doesn't do any validation or prevents other chars, users can type random characters too. 

If we use telephone we will manually have to do validation or prevent this behaviour. Let's keep it number for now if we find any issues we can later switch to telephone.

Reply:
That is the expected call we had taken for phone numbers no? Article - https://technology.blog.gov.uk/2020/02/24/why-the-gov-uk-design-system-team-changed-the-input-type-for-numbers/

Anyway I don't see any case where consumers won't do any validation from their end and since we have validations coming from i18n now, the other characters would be caught in that validation anyways.

Its also rare that someone would try to enter characters because visually it is clear that its phone number input


People might end up accidentally changing the phone number while scrolling because of the default behaviour of type number

https://github.com/razorpay/blade/assets/30949385/d7eab2cc-deb4-46ee-a630-2d10adc23a3f

Reply:
I don't see in the latest preview btw so are we going with telephone?

Reply:
Fine with either one. We can change it to telephone.